selfhost/parser.jakt

```diff
@@ -1108,7 +1108,8 @@ boxed enum ParsedExpression {
                         return false
                     }
                 }
-                if not lhs_values.size() == rhs_values.size() {
```

Comment:
:thinking: interesting that this is allowed here, the typechecker definitely disallows this:
```
> function a() => 42uz
> function b() => 42uz
> not a() == b()
Error: Invalid type for unary operator
```

yet the same expression is allowed in "normal" code...